lib/services/saml.go

```diff
@@ -320,7 +320,7 @@ func (o *SAMLConnectorV2) Equals(other SAMLConnector) bool {
 		if (a.RoleTemplate != nil && b.RoleTemplate == nil) || (a.RoleTemplate == nil && b.RoleTemplate != nil) {
 			return false
 		}
-		if a.RoleTemplate != nil && !a.RoleTemplate.Equals(b.RoleTemplate) {
+		if a.RoleTemplate != nil && !a.RoleTemplate.Equals(b.RoleTemplate.V3()) {
```

Comment:
mm... this (especially if you combine it with 2 lines above) is a very convoluted way to compare two role templates. can't we put all these `if's` behind `RoleTemplate.Equals()` method? BTW you can call it on `nil`.